six/common/stringutils.c

```diff
@@ -48,62 +56,103 @@ char *as_string(PyObject *object)
     return retval;
 }
 
-PyObject *from_yaml(const char *data) {
-    PyObject *retval = NULL;
+int init_stringutils(void) {
     PyObject *yaml = NULL;
-    PyObject *safe_load = NULL;
-
-    if (!data) {
-        goto done;
-    }
+    int ret = EXIT_FAILURE;
 
     char module_name[] = "yaml";
     yaml = PyImport_ImportModule(module_name);
     if (yaml == NULL) {
         goto done;
     }
 
-    char func_name[] = "safe_load";
-    safe_load = PyObject_GetAttrString(yaml, func_name);
-    if (safe_load == NULL) {
+    // get pyyaml load()
+    char load_name[] = "load";
+    yload = PyObject_GetAttrString(yaml, load_name);
+    if (yload == NULL) {
         goto done;
     }
 
-    retval = PyObject_CallFunction(safe_load, "s", data);
+    // We try to use the C-extensions, if they're available, but it's a best effort
+    char c_loader_name[] = "CSafeLoader";
+    loader = PyObject_GetAttrString(yaml, c_loader_name);
+    if (loader == NULL) {
+        PyErr_Clear();
```

Comment:
Should we print something here? Like: "C loader not found, serialization will be slow"

Reply:
Agreed, but I don't think we have any logging mechanism in six...

Reply:
There's nothing straight-forward currently. We could log to stderr, but we'd just get something on the init journal/syslog, but nothing on the actual logs.